### DISCUSSION TITLE
How to use AsyncIOScheduler with gunicorn?

### DISCUSSION THREAD

**Discussion by michal-cech** [agronholm/apscheduler#711] [Q&A]
Hello, is it possible to use AsyncIO Scheduler together with gunicorn? My setup is gunicorn with several unicorn workers running the app. It seems that in this configuration the scheduler never runs, not even once. So far I have tried:

- running on uvicorn instead of gunicorn: this works, but I would like to use gunicorn,
- running with `--preload`: this does not seem to have an effect,
- setting up scheduler in `on_starting` hook function: no avail.

It might be possible to use BackgroundScheduler, but since my scheduling function must use async I have encountered other issues and I would prefer to use AsyncIO scheduler instead.

**Comment by eyadmba** [comment]
I have a similar setup and I'm curious what you mean by running the scheduler _with_ gunicorn. Aren't the scheduler and gunicorn different processes? Is gunicorn playing a different role than an HTTP server here?
